collect_app/src/test/java/org/odk/collect/android/formhierarchy/QuestionAnswerProcessorTest.kt

```diff
@@ -2,25 +2,150 @@ package org.odk.collect.android.formhierarchy
 
 import org.hamcrest.MatcherAssert.assertThat
 import org.hamcrest.Matchers.equalTo
+import org.javarosa.core.model.Constants
 import org.javarosa.core.model.QuestionDef
 import org.javarosa.form.api.FormEntryPrompt
 import org.junit.Test
 import org.mockito.Mockito.mock
 import org.mockito.kotlin.whenever
+import org.odk.collect.android.support.MockFormEntryPromptBuilder
 import org.odk.collect.android.utilities.Appearances
 
 // TODO: Add tests for other question/data types
 class QuestionAnswerProcessorTest {
 
     @Test
     fun noAnswerShouldBeDisplayedForThePrinterWidget() {
-        val prompt = mock<FormEntryPrompt>()
         val question = mock<QuestionDef>()
-        whenever(prompt.question).thenReturn(question)
-        whenever(question.appearanceAttr).thenReturn(Appearances.PRINTER)
+        val prompt = MockFormEntryPromptBuilder()
+            .withQuestion(question)
+            .withAnswerDisplayText("<html>blah</html>>")
+            .withAppearance(Appearances.PRINTER)
+            .build()
 
         val answer = QuestionAnswerProcessor.getQuestionAnswer(prompt, mock(), mock())
 
         assertThat(answer, equalTo(""))
     }
+
+    @Test
+    fun noAnswerShouldBeDisplayedIfItDoesNotExistAndMaskedAppearanceIsUsedForTextAndNumberDataTypes() {
+        listOf(
+            Constants.DATATYPE_TEXT,
+            Constants.DATATYPE_INTEGER,
+            Constants.DATATYPE_DECIMAL
+        ).forEach {
+            val prompt = mock<FormEntryPrompt>()
```

Comment:
This test us still using `mock` directly rather than the `MockFormEntryPromptBuilder` helper.

Reply:
Fixed.